addons/web/static/tests/views/form_tests.js

```diff
@@ -6694,5 +6694,61 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('resequence list lines when discardable lines are present', function (assert) {
+        assert.expect(4);
+
+        var onchangeNum = 0;
+
+        this.data.partner.onchanges = {
+            p: function (obj) {
+                onchangeNum++;
```

Comment:
maybe have a real onchange here (like setting foo to the number of records in the o2m), and then check that the value has been correctly updated?

Reply:
done, I added 4 assertions at each onchange to ensure `foo` is correctly the number of line of `p]